### PR TITLE
docs(guides): typescript-guide rewrite

### PR DESCRIPTION
**Pull request by mtrivera**
## Loader

Removed awesome-ts-loader and moved it to bottom of guide. I think it’s confusing to list two loaders if one is preferred for simplicity, focus on the one that is simple to use. 

## Source Maps

Some people might not know about source maps, so mention another guide that discusses it.

## Using Third Party Libraries

The word third is easier to read than 3rd, and people might not be familiar with its definition. I replaced types github repo with TypeSearch. The search functionality on TypeSearch is easier than reading the documentation on the DefinitelyTyped repo.

Feedback welcome and appreciated!

### CODE REVIEW

File: content/guides/typescript.md [hunk 1]
```diff
@@ -75,20 +76,17 @@ module.exports = {
...
-
-Awesome TypeScript Loader has created a [wonderful explanation](https://github.com/s-panferov/awesome-typescript-loader#differences-between-ts-loader) of the difference between `awesome-typescript-loader` and `ts-loader`.
-
-We chose to use `ts-loader` in this guide as it makes enabling additional webpack features, such as importing other web assets, a bit easier.
+We will use `ts-loader` in this guide as it makes enabling additional webpack features, such as importing other web assets, a bit easier.
 
 
 ## Source Maps
 
+To learn more about Source Maps, see the [development guide](/guides/development.md)
```
Comment (skipjack) [inline content/guides/typescript.md]: Missing period at the end of this sentence.

File: content/guides/typescript.md [hunk 2]
```diff
@@ -136,3 +134,10 @@ declare module "*.svg" {
 ```
 
 Here we declare a new module for SVGs by specifying any import that ends in `.svg` and defining the module's `content` as `any`. We could be more explicit about it being a url by defining the type as string. The same concept applies to other assets including CSS, SCSS, JSON and more.
+
```
Comment (skipjack) [inline content/guides/typescript.md]: We usually use two line breaks between `h2` sections.

File: content/guides/typescript.md [hunk 3]
```diff
@@ -136,3 +134,10 @@ declare module "*.svg" {
...
+## Performance Loader
+
+[`awesome-typescript-loader`](https://github.com/s-panferov/awesome-typescript-loader)
+
+Awesome TypeScript Loader has created a [wonderful explanation](https://github.com/s-panferov/awesome-typescript-loader#differences-between-ts-loader) of the difference between `awesome-typescript-loader` and `ts-loader`. The
+configuration for `awesome-typescript-loader` is more complex than `ts-loader`.
```
Comment (skipjack) [inline content/guides/typescript.md]: This should just be joined with the previous line. At some point we may enforce a max line length (webpack-contrib/webpack-defaults#73) or something to keep it consistent but for now each paragraph should just be on one line for consistency. Then you can turn on word-wrapping in your editor if it's a pain to read.

**Comment by mtrivera** [comment]
I made the corrections

**Comment by skipjack** [comment]
@mtrivera are you interested in synchronizing this with the other guides in a separate PR?

**Comment by mtrivera** [comment]
Sure, but I would need some help with that.  
Are you referring to this from a couple of days ago?

> Also, I was under the impression that we'd be somehow synchronizing this with the other guides (probably modifying the final example from Getting Started would work). This is what we've done for the others in #1258.

**Pull request merged**